### PR TITLE
[SDKS-6268] Create LocalhostSynchronizer

### PR DESCRIPTION
**Pull request by nmayorsplit**

### CODE REVIEW

**Comment by sanzmauro** [comment]
@nmayorsplit the base branch is `main`, it's ok?

**Review by sanzmauro — APPROVED** [review]

File: client/src/main/java/io/split/engine/common/ConsumerSynchronizer.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        _telemetrySyncTask = splitTasks.getTelemetrySyncTask();
+    }
+    @Override
+    public boolean syncAll() {
+        return false;
+    }
+
+    @Override
+    public void startPeriodicFetching() {
+
```
[the remark this answers is not in the document]
Comment (nmayorsplit) [reply client/src/main/java/io/split/engine/common/ConsumerSynchronizer.java]: Thank you, I added it.

File: client/src/main/java/io/split/engine/experiments/SyncTask.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package io.split.engine.experiments;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public interface SyncTask {
+
```
[the remark this answers is not in the document]
Comment (nmayorsplit) [reply client/src/main/java/io/split/engine/experiments/SyncTask.java:4]: I removed it.

File: client/src/main/java/io/split/client/SplitFactoryImpl.java [hunk 3]
```diff
@@ -291,21 +291,18 @@ protected SplitFactoryImpl(String apiToken, SplitClientConfig config, CustomStor
...
-        if (_uniqueKeysTracker != null){
-            try {
-                _uniqueKeysTracker.start();
-            } catch (Exception e) {
-                _log.error("Error trying to init Unique Keys Tracker synchronizer task.", e);
-            }
-        }
+        _telemetrySyncTask = new TelemetrySyncTask(config.get_telemetryRefreshRate(), _telemetrySynchronizer);
+
+        // SyncManager
```
[the remark this answers is not in the document]
Comment (nmayorsplit) [reply client/src/main/java/io/split/client/SplitFactoryImpl.java]: I changed it.

**Review by mredolatti — APPROVED** [review]
LGTM

**Pull request merged**